src/graphene.cpp

```diff
@@ -475,49 +475,45 @@ bool CGrapheneBlock::process(CNode *pfrom,
         for (auto &kv : orphanpool.mapOrphanTransactions)
         {
             uint64_t cheapHash = kv.first.GetCheapHash();
-
-            if (mapPartialTxHash.count(cheapHash)) // Check for collisions
-                collision = true;
-
-            mapPartialTxHash[cheapHash] = kv.first;
+            auto ir = mapPartialTxHash.insert(std::make_pair(cheapHash, kv.first));
+            if (!ir.second)
+                collision = true; // insert returns false if no insertion
```

Comment:
shouldn't this be if (!ir) and not ir.second?

Reply:
https://en.cppreference.com/w/cpp/container/map/insert
overload 2: template< class P > std::pair<iterator,bool> insert( P&& value );